packages/use-query-params/src/index.js

```diff
@@ -0,0 +1,59 @@
+import { parse, stringify } from 'query-string'
+import { useCallback, useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+const useQueryParams = () => {
+  const { location: { search, pathname }, replace } = useHistory()
+  const parseFormat = useCallback(
+    () =>
+      parse(search, {
+        parseNumbers: true,
+        parseBooleans: true,
+        arrayFormat: 'comma',
+      }),
+    [search],
+  )
+  const defaultFnUpdater = useCallback(
+    (currentQueryParams, updater) => ({
+      ...currentQueryParams,
+      ...updater,
+    }),
+    [],
+  )
+  const [state, setState] = useState(parseFormat())
+
+  const setQueryParams = useCallback(
+    updater => {
+      const currentQueryParams = parseFormat()
+      const params =
+        updater instanceof Function
+          ? updater(currentQueryParams)
+          : defaultFnUpdater(currentQueryParams, updater)
+
+      const stringifiedParams = stringify(params, {
+        arrayFormat: 'comma',
+        sort: (a, b) => a.localeCompare(b),
+      })
+
+      if (stringifiedParams !== search.replace('?', '')) {
+        replace(`${pathname}?${stringifiedParams}`)
+
+        setState(params)
+      }
+
+      return state
+    },
+    [parseFormat, defaultFnUpdater, search, state, replace, pathname],
+  )
+
+  useEffect(() => {
+    setQueryParams()
+  }, [setQueryParams])
+
+  return {
```

Comment:
Shouldn't we return an array here to stay consistent with react hooks ?

Reply:
Nevermind, would only be useful if we would use this hook multiple times in the same component

Reply:
I believe it's a good practice if the hook will be use more than one time in a component 

```javascript
So that you can more easily do ...

const [count1, setCount1] = useState(0);
const [count2, setCount2] = useState(0);

With an object you'd need to do ...

const { count: count1, setCount: setCount1 } = useState(0);
const { count: count2, setCount: setCount2 } = useState(0);
```